packages/elements/src/photon-medication-search/photon-medication-search-component.tsx

```diff
@@ -0,0 +1,350 @@
+      return (
+        x.treatment.name.toLowerCase().includes(searchTextLowerCase) ||
+        x.name?.toLowerCase().includes(searchTextLowerCase)
+      );
+    }
+    return x.name.toLowerCase().includes(searchTextLowerCase);
+  });
+}
+
+export const boldSubstring = (inputString: string, substring: any) => {
+  const substrings = substring.split(' ').filter((part: string) => part.length > 0);
+  const regex = new RegExp(`(${substrings.join('|')})`, 'gi');
+  const parts = inputString.split(regex);
+  return parts.map((part) => {
+    if (substrings.some((sub: string) => sub.toLowerCase() === part?.toLowerCase())) {
+      return <strong class="font-extrabold">{part}</strong>;
+    } else {
+      return part;
+    }
+  });
+};
+
+function displayTreatment(
+  t: Treatment | TreatmentOption,
+  groupAccess: boolean,
+  searchText: string
+) {
+  return groupAccess ? (
+    <p class="text-sm whitespace-normal leading-snug my-1">{boldSubstring(t.name, searchText)}</p>
+  ) : (
+    t.name || ''
+  );
+}
+
+function displayPrescriptionTemplate(
+  t: PrescriptionTemplate,
+  groupAccess: boolean,
+  searchText: string
+) {
+  if (groupAccess) {
+    const refills = t.fillsAllowed ? t.fillsAllowed - 1 : 0;
+
+    return (
+      <div class="my-1">
+        <div class="text-sm whitespace-normal font-medium mb-1">
+          {t.name ? <span class="text-blue-600">({boldSubstring(t.name, searchText)}): </span> : ''}
+          {boldSubstring(t.treatment.name, searchText)}
+        </div>
+        <div class="text-xs text-gray-500 truncate">
+          {t.dispenseQuantity} {t.dispenseUnit}, {t.daysSupply}{' '}
+          {t.daysSupply === 1 ? 'day' : 'days'} supply, {refills}{' '}
+          {refills === 1 ? 'refill' : 'refills'}, {t.instructions}
+        </div>
+      </div>
+    );
+  }
+
+  return t.treatment.name;
+}
+
+function getGroupsConfig(props: ComponentProps) {
+  return [
+    {
+      label: 'OFF CATALOG',
+      filter: (t: any) =>
+        t && typeof t === 'object' && props.offCatalogOption && t.id === props.offCatalogOption.id
+    },
+    {
+      label: 'PERSONAL TEMPLATES',
+      filter: (t: any) => t && typeof t === 'object' && 'treatment' in t && t.isPrivate
+    },
+    {
+      label: 'ORGANIZATION TEMPLATES',
+      filter: (t: any) => t && typeof t === 'object' && 'treatment' in t && !t.isPrivate
+    },
+    {
+      label: 'ORGANIZATION CATALOG',
+      filter: (t: any) =>
+        t && typeof t === 'object' && 'name' in t && !('treatment' in t) && !('medicationId' in t)
+    },
+    {
+      label: 'ALL TREATMENTS',
+      filter: (t: any) => t && typeof t === 'object' && 'medicationId' in t
+    }
+  ];
+}
+
+// Component Definition
+
+interface ComponentProps {
+  label?: string;
+  required: boolean;
+  invalid: boolean;
+  helpText?: string;
+  catalogId?: string;
+  disabled: boolean;
+  formName?: string;
+  selected?: Treatment | PrescriptionTemplate | TreatmentOption;
+  offCatalogOption?: Medication;
```

Comment:
Are these actually supposed to be props? How would you pass them in?

Reply:
There's definitely some cleanup to be done here. This is residual. I see it's using the selected id to determine if row item is selected, but that doesn't mean we need the whole selected object in this comp. I can rework that, I'll leave for now tho.